File: soldier.py
class Soldier:
    def __init__(self, r, c):
        self.r = r
        self.c = c

    def numberOfNeighbors(self, board):
        directions = [
            (-1,  0), (-1, +1), ( 0, +1), (+1, +1), 
            (+1,  0), (+1, -1), ( 0, -1), (-1, -1)
        ]
        count = 0
        for dc, dr in directions:
            neighbor_row = self.r + dr
            neighbor_col = self.c + dc
            if 0 <= neighbor_row < 10 and 0 <= neighbor_col < 10:
                if type(board[neighbor_row][neighbor_col]) is Soldier:
                    count += 1
        return count

File: soldier.js
export class Soldier {
    constructor(r, c) {
        this.r = r;
        this.c = c;
    }

    numberOfNeighbors(board) {
        let directions = [
            [-1, 0], [-1, 1], [0, 1], [1, 1],
            [1, 0], [1, -1], [0, -1], [-1, -1]
        ];
        let count = 0;

        directions.forEach(([dr, dc]) => {
            let neighbor_row = this.r + dr;
            let neighbor_col = this.c + dc;
            if (neighbor_row >= 0 && neighbor_row < 10 && neighbor_col >= 0 && neighbor_col < 10) {
                if (board[neighbor_row][neighbor_col] instanceof Soldier) {
                    count += 1;
                }
            }
        });
        
        return count;
    }


}